File: plugins/MediaChannelFix/src/main/java/com/tsq/plugins/MediaChannelFix.java
package com.tsq.plugins;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import com.aliucord.Logger;
import com.aliucord.Utils;
import com.aliucord.annotations.AliucordPlugin;
import com.aliucord.entities.Plugin;
import com.aliucord.patcher.*;
import com.aliucord.utils.DimenUtils;
import com.aliucord.views.Button;
import com.aliucord.views.DangerButton;
import com.discord.api.channel.Channel;
import com.aliucord.widgets.BottomSheet;
import com.aliucord.wrappers.ChannelWrapper;
import com.discord.api.channel.ForumTag;
import com.discord.utilities.rest.RestAPI;
import com.discord.stores.StoreStream;
import com.discord.widgets.forums.ForumPostCreateManager;
import java.lang.reflect.Method;
import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.List;
import com.discord.widgets.chat.MessageManager;
import com.discord.stores.StoreThreadDraft;
import kotlin.jvm.functions.Function2;
import android.content.Context;
import okhttp3.MultipartBody;
import okhttp3.MediaType;
import okhttp3.RequestBody;
import okhttp3.Headers;
import java.util.Iterator;
import androidx.fragment.app.FragmentActivity;
import rx.Observable;
import d0.t.n;
import android.view.Gravity;
import android.graphics.Typeface;
import android.widget.LinearLayout;
import com.discord.app.AppBottomSheet;
import android.view.ViewGroup;
import android.widget.TextView;
import android.view.LayoutInflater;

@AliucordPlugin
public class MediaChannelFix extends Plugin {
	private final Logger logger = new Logger("MediaChannelFix");
	private final List<Long> selectedTagIds = new ArrayList<>();
	private boolean isReinvoked = false;
	private Object[] capturedSendArgs;
	private boolean isSendingManually = false;
	private String bonmun;
	private String name;

	@Override
	public void start(Context context) throws NoSuchMethodException {
		patcher.patch(Channel.class, "D", new Class<?>[]{}, new Hook(param -> {
			if ((int) param.getResult() == 16) {
				param.setResult(15);
			}
		}));
	}


	@Override
	public void stop(Context context) { patcher.unpatchAll(); }
}

File: plugins/MediaChannelFix/src/main/kotlin/com/tsq/plugins/MediaChannelFix.kt
package com.tsq.plugins;

import android.content.Context
import android.widget.ImageView
import android.content.res.Resources
import androidx.core.content.res.ResourcesCompat

import com.aliucord.annotations.AliucordPlugin
import com.aliucord.entities.Plugin
import com.aliucord.patcher.*
import com.aliucord.Utils
import com.aliucord.utils.ReflectUtils

import com.discord.api.channel.Channel
import com.discord.databinding.WidgetChannelsListItemChannelBinding
import com.discord.widgets.channels.list.WidgetChannelsListAdapter
import com.discord.widgets.channels.list.items.ChannelListItem
import com.discord.widgets.channels.list.items.ChannelListItemTextChannel

@AliucordPlugin
class MediaChannelFix: Plugin() {
	private val isMine = ThreadLocal.withInitial { false }
	
	override fun start(context: Context) {
		val getTypeMethod by lazy { Channel::class.java.getDeclaredMethod("D") }
		patcher.patch(getTypeMethod, Hook { param ->
			val rawType = param.result as Int
			if (rawType == 16) {
				if (isMine.get() == true) { 
					param.result = 16 
				} else {
					param.result = 15 
				}
			}
		})
		
		val onConfigureMethod by lazy { WidgetChannelsListAdapter.ItemChannelText::class.java.getDeclaredMethod("onConfigure", Int::class.javaPrimitiveType, ChannelListItem::class.java) }
		patcher.patch(onConfigureMethod, Hook { param ->
			try {
				val channelItem = param.args[1] as ChannelListItemTextChannel
				isMine.set(true)
				val chType = channelItem.getChannel().D() // getType()
				isMine.set(false)

				if (chType == 16) {
					val _this = param.thisObject as WidgetChannelsListAdapter.ItemChannelText
					val binding = ReflectUtils.getField(_this, "binding") as WidgetChannelsListItemChannelBinding
					val channelIcon = binding.getRoot().findViewById(Utils.getResId("channels_item_channel_hash", "id")) as ImageView
						
					val customIconId = resources?.getIdentifier("media_channel", "drawable", "com.tsq.plugins")
						
					if (customIconId != 0) {
						val vectorDrawable = ResourcesCompat.getDrawable(resources!!, customIconId!!, null)
						if (vectorDrawable != null) {
							channelIcon.setImageDrawable(vectorDrawable)
						}
					}
				}
			} catch (e: Throwable) {
				logger.error("mediaChIcon", e)
			}
		})
	}

	override fun stop(context: Context) { patcher.unpatchAll() }
}
